contracts/contracts/coordination/Coordinator.sol

```diff
@@ -124,6 +133,28 @@ contract Coordinator is AccessControlDefaultAdminRules {
         _setRoleAdmin(INITIATOR_ROLE, bytes32(0));
     }
 
+    function setProviderPublicKey(BLS12381.G1Point calldata _publicKey) public {
+        uint32 lastRitualId = uint32(rituals.length);
+        address provider = application.stakingProviderFromOperator(msg.sender);
+
+        ParticipantKey memory newRecord = ParticipantKey(lastRitualId, _publicKey);
+        keysHistory[provider].push(newRecord);
+
+        emit ParticipantPublicKeySet(provider, _publicKey);
+    }
+
+    function getProviderPublicKey(address _address, uint _ritualId) public view returns (BLS12381.G1Point memory) {
```

Comment:
If this function is not called by the other functions in the contract i.e. only called from an external entitiy, I believe using `external` instead saves a copy step of arguments.

Also, let's call the parameter `_provider`.

```suggestion
    function getProviderPublicKey(address _provider, uint _ritualId) external view returns (BLS12381.G1Point memory) {
```

Reply:
:heavy_check_mark: